src/components/app.js

```diff
@@ -3,14 +3,33 @@ import Header from './header';
 import Footer from './footer';
 import CarsList from './cars/carslist';
 
-export default class App extends Component{
+import { connect } from 'react-redux';
+import setCarsAction from '../actions/actionCars'
+
+class App extends Component{
 	render(){
 		return(
 			<div>
 				<Header />
-				<CarsList />
+				<CarsList cars={this.props.cars} setCars={this.props.setCarsFunction} />
 				<Footer />
 			</div>
 		);
 	}
 }
+
+function mapStateToProps(state){
+	return {
+		cars: state.carsList.cars
+	}
+}
+
+function mapDispatchToProps(dispatch) {
+	return {
+		setCarsFunction: cars => {
+			dispatch(setCarsAction(cars))
+		}
+	}
+}
+
+export default connect(mapStateToProps, mapDispatchToProps)(App);
```

Comment:
One of the main advantages of redux is:
You don't need to connect it into root component
You can connect it wherever you need it.
So you don't need cars in app component. You need it in CarList.
Remove all this to CarList.